src/auth/auth-api-request.ts

```diff
@@ -1440,7 +1510,118 @@ export class AuthRequestHandler extends AbstractAuthRequestHandler {
+   */
+  public getTenant(tenantId: string): Promise<TenantServerResponse> {
+    if (!validator.isNonEmptyString(tenantId)) {
+      return Promise.reject(new FirebaseAuthError(AuthClientErrorCode.INVALID_TENANT_ID));
+    }
+    return this.invokeRequestHandler(this.tenantMgmtResourceBuilder, GET_TENANT, {}, {tenantId})
+      .then((response: any) => {
+        return response as TenantServerResponse;
+      });
+  }
+
+  /**
+   * Exports the tenants (single batch only) with a size of maxResults and starting from
+   * the offset as specified by pageToken.
+   *
+   * @param {number=} maxResults The page size, 1000 if undefined. This is also the maximum
+   *     allowed limit.
+   * @param {string=} pageToken The next page token. If not specified, returns tenants starting
+   *     without any offset. Tenants are returned in the order they were created from oldest to
+   *     newest, relative to the page token offset.
+   * @return {Promise<object>} A promise that resolves with the current batch of downloaded
+   *     tenants and the next page token if available. For the last page, an empty list of tenants
+   *     and no page token are returned.
+   */
+  public listTenants(
+      maxResults: number = MAX_LIST_TENANT_PAGE_SIZE,
+      pageToken?: string): Promise<{tenants: TenantServerResponse[], nextPageToken?: string}> {
+    const request = {
+      pageSize: maxResults,
+      pageToken,
+    };
+    // Remove next page token if not provided.
+    if (typeof request.pageToken === 'undefined') {
+      delete request.pageToken;
+    }
+    return this.invokeRequestHandler(this.tenantMgmtResourceBuilder, LIST_TENANTS, request)
+        .then((response: any) => {
+          if (!response.tenants) {
+            response.tenants = [];
+            delete response.nextPageToken;
+          }
+          return response as {tenants: TenantServerResponse[], nextPageToken?: string};
+        });
+  }
+
+  /**
+   * Deletes a tenant identified by a tenantId.
+   *
+   * @param {string} tenantId The identifier of the tenant to delete.
+   * @return {Promise<void>} A promise that resolves when the tenant is deleted.
+   */
+  public deleteTenant(tenantId: string): Promise<void> {
+    if (!validator.isNonEmptyString(tenantId)) {
+      return Promise.reject(new FirebaseAuthError(AuthClientErrorCode.INVALID_TENANT_ID));
+    }
+    return this.invokeRequestHandler(this.tenantMgmtResourceBuilder, DELETE_TENANT, {}, {tenantId})
+      .then((response: any) => {
+        // Return nothing.
+      });
+  }
+
+  /**
+   * Creates a new tenant with the properties provided.
+   *
+   * @param {TenantOptions} tenantOptions The properties to set on the new tenant to be created.
+   * @return {Promise<TenantServerResponse>} A promise that resolves with the newly created tenant object.
+   */
+  public createTenant(tenantOptions: TenantOptions): Promise<object> {
```

Comment:
Need to check whether type (LIGHTWEIGHT, FULL_SERVICE) is always specified

Reply:
Thanks for reviewing. I actually validate that. It was done in my first PR which you didn't get a chance to review.
Check the line below:
`request = Tenant.buildServerRequest(tenantOptions, true);`

This will pass `createRequest` set to true:
https://github.com/firebase/firebase-admin-node/blob/temp-multi-tenancy/src/auth/tenant.ts#L76

This will trigger this check:
https://github.com/firebase/firebase-admin-node/blob/temp-multi-tenancy/src/auth/tenant.ts#L149
For create requests, the type has to be passed and has to be set to `lightweight` or `full_service`.

I have added a check below in the unit tests in `createTenant() should be rejected given valid parameters with no type`
https://github.com/firebase/firebase-admin-node/blob/temp-multi-tenancy/test/unit/auth/auth-api-request.spec.ts#L3814